xmipp

```diff
@@ -655,18 +684,50 @@ def checkCompiler(configDict):
 
 def configMPI(configDict):
     mpiDetected = []
-    if configDict["MPI_CC"] == "" and checkProgram("mpicc"):
-        configDict["MPI_CC"]="mpicc"
-        mpiDetected.append('mpicc')
-    if configDict["MPI_CXX"] == "" and checkProgram("mpiCC"):
-        configDict["MPI_CXX"]="mpiCC"
-        mpiDetected.append('mpiCC')
+    if configDict["MPI_RUN"] == "":
+        if checkProgram("mpirun", False):
+            configDict["MPI_RUN"] = "mpirun"
+            mpiDetected.append('mpirun')
+        elif checkProgram("mpiexec", False):
+            configDict["MPI_RUN"] = "mpiexec"
+            mpiDetected.append('mpiexec')
+        else:
+            mpiDir = findFileInDirList('mpirun', ['/usr/lib/openmpi/bin',
```

Comment:
One thing is to help them, other is to do something for them without them being able to say anything.
Like this, you don't give any flexibility to advanced users, because you set whatever is in PATH or other locations, not giving them any chance to protest. As for Average Joes, how many of them use Centos? If they do, it's probably because it's a shared machine. In such an environment, you cannot assume anything, and it's typically admins responsibility to install software. Typically,  modules have to be activated, and all that can and will result in a pretty messy environment.
I understand why you think your approach is good. I'm not saying that it's a bad idea, but that it will lead to version clashes and really hard to find runtime and compilation errors.

If we get 'a lot' of mails asking how to configure things, we should consider how good is our documentation.

Reply:
Ok, got it. If nothing provided but something found, I will ask by giving as default what is found.

Do you agree on that?

On the other hand, note that all environ variables needed in runtime are stored in a json/dict in the build (and took by xmipp plugin).